src/thing/Portal.cpp

```diff
@@ -109,32 +109,24 @@ void Portal::Start() {
   });
 
   server_.on("/config", [&] () {
-    DynamicJsonBuffer jsonBuffer;
     if (server_.method() == HTTP_GET) {
-      JsonObject& root = jsonBuffer.createObject();
-      root["host"] = config_.host.c_str();
-      root["auth"] = config_.auth.c_str();
-      root["path"] = config_.path.c_str();
-      root["wifi_ssid"] = config_.wifi_ssid.c_str();
-      root["wifi_key"] = config_.wifi_key.c_str();
+      auto client = server_.client();
```

Comment:
We do, it is in the "size" callback below (117). We need the size in order to correctly set the content length header. The way this ends up is not ideal from a readability standpoint but we can't get the size before building the json object and we don't need the size for writing to a file. So we could add another level of abstraction and introduce a ConfigSerializer object that you can prime with the config, get the size, then write to a file. Or do this callback inverted flow dance here. The other object doesn't actually sound that bad now that I think it through but I am not sure it is worth changing right now.